### ISSUE TITLE
[Feature]: Using connect Timeout in Router: httpx.Timeout

### ISSUE THREAD

**Issue by Ludwig-Graef** [BerriAI/litellm#3162]
### The Feature

I need to be able to wait a certain amount of time before I get a connection timeout.

### Motivation, pitch

I've done this before:
`AsyncAzureOpenAI(timeout=httpx.Timeout(timeout=300.0, connect=20.0), **kwargs)`.
See httpx.Timeout details here: https://www.python-httpx.org/advanced/timeouts/

I now want to use it as a parameter in the litellm router.
Is there any way to get similar behaviour?

**Comment by m-revetria** [comment]
Hi, I came here because I'm having a slightly different but related issue.

In my case I need to pass other parameters to the `httpx`, due to a temporary issue with SSL certificates I need to disable ssl check for now. We used to create `openai.AzureOpenAI` as `AzureOpenAI(http_client=httpx.Client(http2=True, verify=False))`

This feature might be more generic in the sense of allowing users to customize any aspect of the `httpx` client used.

Btw, my project is making use of the `Router` and `ChatLiteLLMRouter` classes and couldn't find a way to inject the client so far. Any workaround would be really appreciated

**Comment by ghost** [comment]
hey @m-revetria @Ludwig-Graef acknowledging this, i believe this is already supported. let me qa and get back to you

**Comment by ghost** [comment]
Hey @Ludwig-Graef just pushed a fix for this - https://github.com/BerriAI/litellm/commit/7730520fb03f97f06203ff6b12e8284a93e065ac 

should be live soon. Here's how to set it 

Curious - would you want a hosted proxy to handle the loadbalancing for you?

```python 
from litellm import Router 
import httpx, os 

os.environ["AZURE_API_KEY"] = "my-key"
os.environ["AZURE_API_BASE"] = "my-api-base"
os.environ["AZURE_API_VERSION"] = "my-api-version"

router = Router(
    model_list = [
        {
            "model_name": "test-model", 
            "litellm_params": {
                "model": "azure/chatgpt-v-2",
                "api_key": os.getenv("AZURE_API_KEY"), 
                "api_base": os.getenv("AZURE_API_BASE"),
                "api_version": os.getenv("AZURE_API_VERSION"),
                "timeout": httpx.Timeout(timeout=300.0, connect=20.0)
            }
        }
    ]
)
```

**Comment by ghost** [comment]
@m-revetria just pushed the fix - https://github.com/BerriAI/litellm/commit/180718c33f5e688b24098155b92149e862e9935a

You can set it like this: 

```python
litellm.ssl_verify = False
```

or if you're on the proxy: 

```yaml
litellm_settings:
    ssl_verify: False
```


Should be live soon in the next release - v`1.35.30+`. Would you want a hosted proxy to handle load balancing / calling the llm api's?

**Comment by Ludwig-Graef** [comment]
@krrishdholakia 
I have tried to run the code. I ran into these problems:

1. In main.py (line 638): `timeout = float(timeout)` -> This breaks if the timeout is not set as float
2. in bedrock.py (line 595): `config = boto3.session.Config(connect_timeout=timeout, read_timeout=timeout)` -> This also breaks. We would need a check for the type and set it accordingly. 

These two are the only ones that I have tested. We might should add a test for this
